alf/algorithms/diayn_algorithm.py

```diff
@@ -108,37 +119,40 @@ def train_step(self,
         observation, skill = observations_aug
         prev_skill = state
 
+        # normalize observation for easier prediction
+        if self._observation_normalizer is not None:
+            observation = self._observation_normalizer.normalize(observation)
+
         if self._encoding_net is not None:
-            feature, _ = self._encoding_net(observation)
+            feature = self._encoding_net(observation)
```

Comment:
Let's make the interface of all kinds of Network same: always have state_spec property, and forward() always have state argument, and return (output, state) tuple. The reason is to make switching between RNN and non-RNN network easier.